dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreCollectionQueryBuilder.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Cosmos;
+using Microsoft.SemanticKernel.Data;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureCosmosDBNoSQL;
+
+/// <summary>
+/// Contains helpers to build queries for Azure CosmosDB NoSQL.
+/// </summary>
+internal static class AzureCosmosDBNoSQLVectorStoreCollectionQueryBuilder
+{
+    private const string SelectClauseDelimiter = ",";
+    private const string AndConditionDelimiter = " AND ";
+    private const string OrConditionDelimiter = " OR ";
+
+    /// <summary>
+    /// Builds <see cref="QueryDefinition"/> to get items from Azure CosmosDB NoSQL using vector search.
+    /// </summary>
+    public static QueryDefinition BuildSearchQuery<TVector>(
+        TVector vector,
+        List<string> fields,
+        Dictionary<string, string> storagePropertyNames,
+        string vectorPropertyName,
+        string scorePropertyName,
+        VectorSearchOptions searchOptions)
+    {
+        const string VectorVariableName = "@vector";
+        const string OffsetVariableName = "@offset";
+        const string LimitVariableName = "@limit";
+
+        var tableVariableName = AzureCosmosDBNoSQLConstants.TableQueryVariableName;
+
+        var fieldsArgument = fields.Select(field => $"{tableVariableName}.{field}");
+        var vectorDistanceArgument = $"VectorDistance({tableVariableName}.{vectorPropertyName}, {VectorVariableName})";
+        var vectorDistanceArgumentWithAlias = $"{vectorDistanceArgument} AS {scorePropertyName}";
+
+        var selectClauseArguments = string.Join(SelectClauseDelimiter, [.. fieldsArgument, vectorDistanceArgumentWithAlias]);
+
+        var filter = BuildSearchFilter(searchOptions.Filter, storagePropertyNames);
+
+        var filterQueryParameters = filter?.QueryParameters;
+        var filterWhereClauseArguments = filter?.WhereClauseArguments;
+
+        var whereClause = filterWhereClauseArguments is { Count: > 0 } ?
+            $"WHERE {string.Join(AndConditionDelimiter, filterWhereClauseArguments)}" :
+            string.Empty;
+
+        var query =
+            $"SELECT {selectClauseArguments} " +
+            $"FROM {tableVariableName} " +
+            $"{whereClause} " +
+            $"ORDER BY {vectorDistanceArgument} " +
+            $"OFFSET {OffsetVariableName} LIMIT {LimitVariableName} ";
```

Comment:
Any concerns here about SQL injection attacks? Do we have a method to protect against these?

Reply:
Anything that can come from end users is injected into query using query parameters - `QueryDefinition.WithParameter()`, which should prevent from SQL injection attacks.